test/Microsoft.Health.Fhir.Tests.E2E/Rest/CompartmentTests.cs

```diff
@@ -0,0 +1,94 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using Hl7.Fhir.Model;
+using Microsoft.Health.Fhir.Tests.Common;
+using Microsoft.Health.Fhir.Tests.E2E.Common;
+using Microsoft.Health.Fhir.Web;
+using Xunit;
+using Task = System.Threading.Tasks.Task;
+
+namespace Microsoft.Health.Fhir.Tests.E2E.Rest.Search
+{
+    public class CompartmentTests : SearchTestsBase<HttpIntegrationTestFixture<Startup>>
+    {
+        private readonly Patient _patient;
+        private readonly Observation _observation;
+        private readonly Device _device;
+        private readonly DeviceComponent _deviceComponent;
+        private readonly Encounter _encounter;
+        private readonly Condition _condition;
+
+        public CompartmentTests(HttpIntegrationTestFixture<Startup> fixture)
+            : base(fixture)
+        {
+            // Delete all relevant resources before starting the test.
+            Client.DeleteAllResources(ResourceType.Device, "_id=d1").Wait();
+            Client.DeleteAllResources(ResourceType.DeviceComponent, "_id=example").Wait();
+            Client.DeleteAllResources(ResourceType.Patient, "_id=f001").Wait();
```

Comment:
All the other tests are using the id generated by the server. Is there any particular reason these needs to be client id? Also, it will be much easier if you just use tags since you could just delete everything matching the tag as opposed to specifying the id and deleting individual items.